Source/Scene/Scene.js

```diff
@@ -3043,6 +3046,21 @@ define([
      * @exception {DeveloperError} Picking from the depth buffer is not supported. Check pickPositionSupported.
      */
     Scene.prototype.pickPosition = function(windowPosition, result) {
+
+        if(defined(windowPosition) && pickPositionCache.hasOwnProperty(windowPosition.toString())){
+            var cached = pickPositionCache[windowPosition.toString()];
```

Comment:
You may want to add a comment somewhere that the cached result could be `undefined`. This way it is clear to whoever is reading that `hasOwnProperty` is used instead of just `defined`.